xbmc/ServiceManager.cpp

```diff
@@ -86,6 +90,9 @@ void CServiceManager::DeinitTesting()
   m_binaryAddonManager.reset();
   m_addonMgr.reset();
   m_databaseManager.reset();
+#if !defined(TARGET_WINDOWS) && defined(HAS_FILESYSTEM_SMB)
+  m_WSDiscovery.reset();
```

Comment:
Can you remove the ifguards? `m_WSDiscovery.reset();` will do the correct thing always.

Reply:
Will do tomorrow - after Jenkins has completed his build.

Reply:
Done.

Reply:
Have to revert the change. m_WSDiscovery does not exist on all platforms.